python-project/mortar/image/threshold.py

```diff
@@ -16,25 +15,17 @@
 class Threshold(Filter):
     """
     Perform thresholding on an image.
-
-    TODO this class doesn't yet support the necessary thresholding options.
     """
     name = 'Threshold'
 
-    def run(self, input: PILImage) -> Optional[PILImage]:
+    def run(self,
+            input: PILImage,
+            threshval: float = 127,
+            maxval: float = 255,
+            invert: bool = False) -> Optional[PILImage]:
         img = np.array([input.getdata()], dtype='uint8')
-        ret, thresh1 = cv.threshold(img, 127, 255, cv.THRESH_BINARY)
-        ret, thresh2 = cv.threshold(img, 127, 255, cv.THRESH_BINARY_INV)
-        ret, thresh3 = cv.threshold(img, 127, 255, cv.THRESH_TRUNC)
-        ret, thresh4 = cv.threshold(img, 127, 255, cv.THRESH_TOZERO)
-        ret, thresh5 = cv.threshold(img, 127, 255, cv.THRESH_TOZERO_INV)
-        titles = ['Original Image', 'BINARY', 'BINARY_INV', 'TRUNC', 'TOZERO',
-                  'TOZERO_INV']
-        images = [img, thresh1, thresh2, thresh3, thresh4, thresh5]
-        for i in range(6):
-            plt.subplot(2, 3, i + 1)
-            plt.imshow(images[i], 'gray', vmin=0, vmax=255)
-            plt.title(titles[i])
-            plt.xticks([]), plt.yticks([])
-
-        return PIL.Image.frombytes('L', input.size, bytes(thresh1))
+
+        thresh_type = cv.THRESH_BINARY_INV if invert else cv.THRESH_BINARY
+        ret, thresh = cv.threshold(img, threshval, maxval, thresh_type)
```

Comment:
`ret` would be the computed thresholding value. According to the OpenCV docs for this function, that value isn't even calculated for the binary/binary inverted thresholding types, so `ret` isn't something we'll use.

It's idiomatic to name "throwaway" variables using a single underscore, so I'd recommend doing that with `ret` in this case. See https://stackoverflow.com/questions/5893163/what-is-the-purpose-of-the-single-underscore-variable-in-python